rollout/trafficrouting.go

```diff
@@ -171,6 +171,13 @@ func (c *rolloutContext) reconcileTrafficRouting() error {
 				return err
 			}
 
+			// If we are aborting, we want to set the desired weight to 0, but we also want to set the canary service
```

Comment:
if we are setting desired weight to be zero doesnt mean that we need to point canary service to point to stable service right. If canary replica count is zero ( replica set doesnt exist), then point canary service back to stable service

Reply:
Made a change to support dynamic stable scale on and off, this whole PR still needs e2e tests though